NOnion.Tests/MonohopCircuits.cs

```diff
@@ -70,5 +73,49 @@ public async Task CanReceiveCompressedConsensusOverMonohopCircuit()
             Assert.That(response.Contains("network-status-version"));
         }
 
+        private async Task<ServerDescriptorEntry> GetRouter ()
+        {
+            using TorGuard guard = await TorGuard.NewClientAsync(torServer);
+            TorCircuit circuit = new(guard);
+            TorStream stream = new(circuit);
+
+            await circuit.CreateAsync(FSharpOption<CircuitNodeDetail>.None);
+            await stream.ConnectToDirectoryAsync();
+
+            var httpClient = new TorHttpClient(stream, torServer.Address.ToString());
+            var serverDescriptors = ServerDescriptorsDocument.Parse(await httpClient.GetAsStringAsync("/tor/server/all", false));
+
+            return serverDescriptors.Routers.First(x => FSharpOption<string>.get_IsSome(x.NTorOnionKey) && FSharpOption<string>.get_IsSome(x.Fingerprint) && FSharpOption<int>.get_IsSome(x.DirectoryPort) &&  x.DirectoryPort.Value != 0);
+        }
+
+        [Test]
+        public async Task CanReceiveCompressedConsensusOverNonFastMonohopCircuit()
+        {
+            ServerDescriptorEntry server = null;
+            try
+            {
+                server = await GetRouter();
+            }
+            catch
+            {
+                Assert.Inconclusive();
```

Comment:
why this?

Reply:
Because that part of the test is not related to the test itself, it's just so we can find a router and the info for that router to connect to (like how we connect to consensus in real world and cache the servers etc) which is the important part of the test

Reply:
but what exception would happen if it can't get a router? I don't like generic catches